---
bibliography: [/Users/mdozmorov/Documents/Work/VCU_grants/1_Dozmorov/education.bib]
csl: /Users/mdozmorov/Documents/Work/presentations/Templates/manuscript_template/styles.ref/bioinformatics.csl
output:
  word_document:
    reference_docx: /Users/mdozmorov/Documents/Work/presentations/Templates/manuscript_template/styles.doc/Arial_8_single_space_narrow_margins_landscape.docx
  pdf_document: default
  html_document: default
---

```{r setup, echo=FALSE, message=FALSE, warning=FALSE}
# Set up the environment
library(knitr)
opts_chunk$set(cache.path='cache/', fig.path='img/', cache=T, tidy=T, fig.keep='high', echo=F, dpi=100, warnings=F, message=F, comment=NA, warning=F, results='as.is', fig.width = 10, fig.height = 6) #out.width=700, 
library(pander)
panderOptions('table.split.table', Inf)
set.seed(1)
library(dplyr)
options(stringsAsFactors = FALSE)
```

```{r}
library(httr)
library(jsonlite)
source("scripts/utils.R")

url <- "https://api.github.com"
```


### Table 1. Popular collections of bioinformatics resources, accessed on `r Sys.Date()`

```{r}
# General resources
path <- c("repos/greenelab/deep-review", "repos/hussius/deeplearning-biology", "repos/danielecook/Awesome-Bioinformatics", "repos/jtleek/genomicspapers", "repos/shenwei356/awesome", "repos/jdidion/biotools", "repos/crazyhottommy/getting-started-with-genomics-tools-and-resources")
mtx1 <- get_github_stats(url = url, path = path)
```

```{r}
# Field-specific resources
path <- c("repos/seandavi/awesome-single-cell", "repos/crazyhottommy/RNA-seq-analysis", "repos/crazyhottommy/ChIP-seq-analysis", "repos/seandavi/awesome-cancer-variant-databases", "repos/crazyhottommy/DNA-seq-analysis", "repos/johandahlberg/awesome-10x-genomics", "repos/stevetsa/awesome-microbes", "repos/crazyhottommy/DNA-methylation-analysis")
mtx2 <- get_github_stats(url = url, path = path)
```

```{r}
# Add headers, combine
header1 <- data.frame(matrix(ncol = ncol(mtx1), nrow = 1))
colnames(header1) <- colnames(mtx1)
header1$Name <- "General bioinformatics collections"
header2 <- data.frame(matrix(ncol = ncol(mtx2), nrow = 1))
colnames(header2) <- colnames(mtx2)
header2$Name <- "Field-specific bioinformatics collections"
mtx <- rbind(header1, mtx1, header2, mtx2)
```

```{r}
kable(mtx)
```

**Supplementary Table 1. Select data science resources.**  Metrics in all tables were assessed on `r Sys.Date()`.

```{r}
path <- c("repos/qinwf/awesome-R", "repos/veggiemonk/awesome-docker", "repos/pditommaso/awesome-pipeline", "repos/grabear/awesome-rshiny", "repos/sdmg15/Best-websites-a-programmer-should-visit", "repos/mtdvio/every-programmer-should-know", "repos/EbookFoundation/free-programming-books", "repos/awesomedata/awesome-public-datasets")
mtx <- get_github_stats(url = url, path = path)
kable(mtx)
```

**Supplementary Table 2. Examples of lists of lists of computer science and machine learning resources.**

```{r}
path <- c("repos/sindresorhus/awesome", "repos/bayandin/awesome-awesomeness", "repos/emijrp/awesome-awesome", "repos/josephmisiti/awesome-machine-learning", "repos/prakhar1989/awesome-courses", "repos/ChristosChristofidis/awesome-deep-learning", "repos/mlr-org/mlr")
mtx <- get_github_stats(url = url, path = path)
kable(mtx)
```

**Supplementary Table 3. Impact metrics of popular bioinformatics tools and resources.** Only software that is being developed on GitHub, has over 50 stars, and published in peer-review journals was selected.

```{r}
# Generated by "figures_impact.Rmd"
fileNameOut <- "tables/Table_software_stats.csv"
mtx <- read.csv(fileNameOut)
kable(mtx)
```

